### PR TITLE
Filter: fixing after refactoring

### PR DESCRIPTION
**Pull request by michalpristas**
**What is the problem I am trying to address?**

After we went away from workers and redesigned data flow, filters stayed one step behind.
This PR brings them even and adds a bit different meaning to filtering keywords considering new data flow.

**How is the fix applied?**

Change is following
- Include:
  - Treats module as it should be without using filter functionality used to reset parent behavior change
- Exclude:
  - Results in exclusion of the module entirely
- Direct:
  - Goes to the upstream proxy without consulting a local storage
  - Before: went to VCS  directly without consulting Olympus as a backup, treated module as a private repo

Filtering middleware moved after auth. So we don't tell whether we treat or we don't particular modules for unauthenticated users.

Removed handling of pseudo versions, it is treated just as a module rule tells it to

### CODE REVIEW

File: pkg/middleware/filter.go [hunk 1]
```diff
@@ -50,10 +44,6 @@ func NewFilterMiddleware(mf *module.Filter, registryEndpoint string) buffalo.Mid
 	}
 }
 
-func isPseudoVersion(version string) bool {
-	return strings.HasPrefix(version, "v0.0.0-")
-}
-
-func redirectToRegistryURL(registryEndpoint string, u *url.URL) string {
+func redirectToUpstreamURL(registryEndpoint string, u *url.URL) string {
```
Comment (marpio) [inline pkg/middleware/filter.go]: registryEndpoint -> upstreamEndpoint

File: cmd/proxy/actions/app.go [hunk 2]
```diff
@@ -137,6 +129,14 @@ func App(conf *config.Config) (*buffalo.App, error) {
 		app.Use(basicAuth(user, pass))
 	}
 
+	if !conf.FilterOff() {
```
Comment (marpio) [inline cmd/proxy/actions/app.go]: Wouldn't we validate a module even if it's on the `Exclude` list if the `ValidationMiddleware` is added **before** the `FilterMiddleware`?

Comment (michalpristas) [reply cmd/proxy/actions/app.go]: that;s right

Comment (michalpristas) [reply cmd/proxy/actions/app.go]: @marpio changed the order, please check if it makes sense for you now

File: pkg/module/filterRule.go [hunk 3]
```diff
@@ -4,12 +4,13 @@ package module
 type FilterRule int
 
 const (
-	// Default filter rule does not alter default behavior
+	// Default filter rule does not alter default/parent behavior
```
[the remark this answers is not in the document]
Comment (michalpristas) [reply pkg/module/filterRule.go]: nono default is important for behavior, not for configuration. default means that it has a behavior of a parent for example, for configuration `- github.com/a`
checking module module[github.com/a/b] will yield default which means look to its parent.
maybe logic can be rewritten, but i would keep it as a separate PR

[the remark this answers is not in the document]
Comment (manugupt1) [reply pkg/module/filterRule.go]: ahh.. okay! so like a tree.

**Comment by manugupt1** [comment]
@fedepaol Does this look okay to you? Once you give a 👍 , will merge it in.

**Pull request merged**